// bs-webapi/src/Webapi/Webapi__Dom.bs.js
'use strict';

var Webapi__Dom__Types = require("./Dom/Webapi__Dom__Types.bs.js");

var $$AnimationEvent;

var $$Attr;

var $$BeforeUnloadEvent;

var CdataSection;

var $$CharacterData;

var $$Comment;

var CssStyleDeclaration;

var $$ClipboardEvent;

var $$CloseEvent;

var $$CompositionEvent;

var $$CustomEvent;

var $$Document;

var $$DocumentFragment;

var $$DocumentType;

var DomImplementation;

var DomRect;

var DomStringMap;

var DomTokenList;

var $$DragEvent;

var $$Element;

var $$ErrorEvent;

var $$Event;

var $$EventTarget;

var $$FocusEvent;

var $$History;

var HtmlCollection;

var HtmlDocument;

var HtmlElement;

var HtmlFormElement;

var HtmlImageElement;

var HtmlInputElement;

var IdbVersionChangeEvent;

var $$Image;

var $$InputEvent;

var $$KeyboardEvent;

var $$Location;

var $$MouseEvent;

var $$MutationObserver;

var $$MutationRecord;

var $$NamedNodeMap;

var $$Node;

var $$NodeFilter;

var $$NodeIterator;

var $$NodeList;

var $$PageTransitionEvent;

var $$PointerEvent;

var $$PopStateEvent;

var $$ProcessingInstruction;

var $$ProgressEvent;

var $$Range;

var RelatedEvent;

var $$Selection;

var $$ShadowRoot;

var $$StorageEvent;

var SvgZoomEvent;

var $$Text;

var TimeEvent;

var $$TouchEvent;

var $$TrackEvent;

var $$TransitionEvent;

var $$TreeWalker;

var UiEvent;

var $$ValidityState;

var WebGlContextEvent;

var $$WheelEvent;

var $$Window;

var encodeCompareHow = Webapi__Dom__Types.encodeCompareHow;

var decodeCompareResult = Webapi__Dom__Types.decodeCompareResult;

var decodeCompatMode = Webapi__Dom__Types.decodeCompatMode;

var encodeContentEditable = Webapi__Dom__Types.encodeContentEditable;

var decodeContentEditable = Webapi__Dom__Types.decodeContentEditable;

var decodeDeltaMode = Webapi__Dom__Types.decodeDeltaMode;

var encodeDesignMode = Webapi__Dom__Types.encodeDesignMode;

var decodeDesignMode = Webapi__Dom__Types.decodeDesignMode;

var encodeDir = Webapi__Dom__Types.encodeDir;

var decodeDir = Webapi__Dom__Types.decodeDir;

var EventPhase = Webapi__Dom__Types.EventPhase;

var encodeFilterAction = Webapi__Dom__Types.encodeFilterAction;

var encodeInsertPosition = Webapi__Dom__Types.encodeInsertPosition;

var encodeModifierKey = Webapi__Dom__Types.encodeModifierKey;

var decodeNodeType = Webapi__Dom__Types.decodeNodeType;

var decodePointerType = Webapi__Dom__Types.decodePointerType;

var decodeReadyState = Webapi__Dom__Types.decodeReadyState;

var decodeShadowRootMode = Webapi__Dom__Types.decodeShadowRootMode;

var decodeVisibilityState = Webapi__Dom__Types.decodeVisibilityState;

var WhatToShow = Webapi__Dom__Types.WhatToShow;

exports.$$AnimationEvent = $$AnimationEvent;
exports.$$Attr = $$Attr;
exports.$$BeforeUnloadEvent = $$BeforeUnloadEvent;
exports.CdataSection = CdataSection;
exports.$$CharacterData = $$CharacterData;
exports.$$Comment = $$Comment;
exports.CssStyleDeclaration = CssStyleDeclaration;
exports.$$ClipboardEvent = $$ClipboardEvent;
exports.$$CloseEvent = $$CloseEvent;
exports.$$CompositionEvent = $$CompositionEvent;
exports.$$CustomEvent = $$CustomEvent;
exports.$$Document = $$Document;
exports.$$DocumentFragment = $$DocumentFragment;
exports.$$DocumentType = $$DocumentType;
exports.DomImplementation = DomImplementation;
exports.DomRect = DomRect;
exports.DomStringMap = DomStringMap;
exports.DomTokenList = DomTokenList;
exports.$$DragEvent = $$DragEvent;
exports.$$Element = $$Element;
exports.$$ErrorEvent = $$ErrorEvent;
exports.$$Event = $$Event;
exports.$$EventTarget = $$EventTarget;
exports.$$FocusEvent = $$FocusEvent;
exports.$$History = $$History;
exports.HtmlCollection = HtmlCollection;
exports.HtmlDocument = HtmlDocument;
exports.HtmlElement = HtmlElement;
exports.HtmlFormElement = HtmlFormElement;
exports.HtmlImageElement = HtmlImageElement;
exports.HtmlInputElement = HtmlInputElement;
exports.IdbVersionChangeEvent = IdbVersionChangeEvent;
exports.$$Image = $$Image;
exports.$$InputEvent = $$InputEvent;
exports.$$KeyboardEvent = $$KeyboardEvent;
exports.$$Location = $$Location;
exports.$$MouseEvent = $$MouseEvent;
exports.$$MutationObserver = $$MutationObserver;
exports.$$MutationRecord = $$MutationRecord;
exports.$$NamedNodeMap = $$NamedNodeMap;
exports.$$Node = $$Node;
exports.$$NodeFilter = $$NodeFilter;
exports.$$NodeIterator = $$NodeIterator;
exports.$$NodeList = $$NodeList;
exports.$$PageTransitionEvent = $$PageTransitionEvent;
exports.$$PointerEvent = $$PointerEvent;
exports.$$PopStateEvent = $$PopStateEvent;
exports.$$ProcessingInstruction = $$ProcessingInstruction;
exports.$$ProgressEvent = $$ProgressEvent;
exports.$$Range = $$Range;
exports.RelatedEvent = RelatedEvent;
exports.$$Selection = $$Selection;
exports.$$ShadowRoot = $$ShadowRoot;
exports.$$StorageEvent = $$StorageEvent;
exports.SvgZoomEvent = SvgZoomEvent;
exports.$$Text = $$Text;
exports.TimeEvent = TimeEvent;
exports.$$TouchEvent = $$TouchEvent;
exports.$$TrackEvent = $$TrackEvent;
exports.$$TransitionEvent = $$TransitionEvent;
exports.$$TreeWalker = $$TreeWalker;
exports.UiEvent = UiEvent;
exports.$$ValidityState = $$ValidityState;
exports.WebGlContextEvent = WebGlContextEvent;
exports.$$WheelEvent = $$WheelEvent;
exports.$$Window = $$Window;
exports.encodeCompareHow = encodeCompareHow;
exports.decodeCompareResult = decodeCompareResult;
exports.decodeCompatMode = decodeCompatMode;
exports.encodeContentEditable = encodeContentEditable;
exports.decodeContentEditable = decodeContentEditable;
exports.decodeDeltaMode = decodeDeltaMode;
exports.encodeDesignMode = encodeDesignMode;
exports.decodeDesignMode = decodeDesignMode;
exports.encodeDir = encodeDir;
exports.decodeDir = decodeDir;
exports.EventPhase = EventPhase;
exports.encodeFilterAction = encodeFilterAction;
exports.encodeInsertPosition = encodeInsertPosition;
exports.encodeModifierKey = encodeModifierKey;
exports.decodeNodeType = decodeNodeType;
exports.decodePointerType = decodePointerType;
exports.decodeReadyState = decodeReadyState;
exports.decodeShadowRootMode = decodeShadowRootMode;
exports.decodeVisibilityState = decodeVisibilityState;
exports.WhatToShow = WhatToShow;
/* No side effect */
