// bs-webapi/src/Webapi/Canvas/Webapi__Canvas__Canvas2d.bs.js
'use strict';

var Pervasives = require("bs-platform/lib/js/pervasives.js");

var Composite = {
  sourceOver: "source-over",
  sourceIn: "source-in",
  sourceOut: "source-out",
  sourceAtop: "source-atop",
  destinationOver: "destination-over",
  destinationIn: "destination-in",
  destinationOut: "destination-out",
  destinationAtop: "destination-atop",
  lighter: "lighter",
  copy: "copy",
  xor: "xor"
};

var LineCap = {
  butt: "butt",
  round: "round",
  square: "square"
};

var LineJoin = {
  round: "round",
  bevel: "bevel",
  miter: "miter"
};

function setStrokeStyle(ctx, param, v) {
  ctx.strokeStyle = v;
  
}

function setFillStyle(ctx, param, v) {
  ctx.fillStyle = v;
  
}

function reifyStyle(x) {
  var isCanvasGradient = (function(x) { return x instanceof CanvasGradient; });
  var isCanvasPattern = (function(x) { return x instanceof CanvasPattern; });
  return [
          typeof x === "string" ? /* String */0 : (
              isCanvasGradient(x) ? /* Gradient */1 : (
                  isCanvasPattern(x) ? /* Pattern */2 : Pervasives.invalid_arg("Unknown canvas style kind. Known values are: String, CanvasGradient, CanvasPattern")
                )
            ),
          x
        ];
}

function fillStyle(ctx) {
  return reifyStyle(ctx.fillStyle);
}

function strokeStyle(ctx) {
  return reifyStyle(ctx.strokeStyle);
}

exports.Composite = Composite;
exports.LineCap = LineCap;
exports.LineJoin = LineJoin;
exports.setStrokeStyle = setStrokeStyle;
exports.setFillStyle = setFillStyle;
exports.reifyStyle = reifyStyle;
exports.fillStyle = fillStyle;
exports.strokeStyle = strokeStyle;
/* No side effect */
